ProjFS.Mac/PrjFSLibTests/JsonWriterTests.mm

```diff
@@ -0,0 +1,225 @@
+    writer.Add("testKey", static_cast<uint64_t>(32));
+    string jsonResult = writer.ToString();
+    XCTAssertTrue(
+        jsonResult == expectedResult,
+        "%s",
+        ("Expected result: " + expectedResult + " Result: " + jsonResult).c_str());
+}
+
+- (void) testAddMultiplePairs {
+    string expectedResult =
+    "{"
+        "\"testKey\":\"testValue\","
+        "\"testInt\":32,"
+        "\"testUInt64\":9223372036854775807,"
+        "\"test2ndString\":\"testValue2\""
+    "}";
+    JsonWriter writer;
+    writer.Add("testKey", "testValue");
+    writer.Add("testInt", 32);
+    writer.Add("testUInt64", 9223372036854775807ULL);
+    writer.Add("test2ndString", "testValue2");
+    string jsonResult = writer.ToString();
+    XCTAssertTrue(
+        jsonResult == expectedResult,
+        "%s",
+        ("Expected result: " + expectedResult + " Result: " + jsonResult).c_str());
+}
+
+- (void) testEscapingCharacters {
+    vector<pair<char, string>> escapedCharacters =
+    {
+        make_pair('\"', "\\\""),
+        make_pair('\\', "\\\\"),
+        make_pair('\n', "\\n"),
+        make_pair('\r', "\\r"),
+        make_pair('\t', "\\t"),
+        make_pair('\f', "\\f"),
+        make_pair('\b', "\\b"),
+        make_pair(static_cast<char>(2), "\\u0002"),
+        make_pair(static_cast<char>(0x13), "\\u0013"),
+        make_pair(static_cast<char>(0x20), " "),
+    };
+
+    for (const pair<char, string>& charPair : escapedCharacters)
+    {
+        string expectedResult =
+        "{"
+            "\"testKey\":\"" + charPair.second + "\""
+        "}";
+    
+        JsonWriter writer;
+        writer.Add("testKey", string(1, charPair.first));
+        string jsonResult = writer.ToString();
+        
+        XCTAssertTrue(
+            jsonResult == expectedResult,
+            "%s",
+            ("Expected result: " + expectedResult + " Result: " + jsonResult).c_str());
+    }
+}
+
+- (void) testMultipleEscapedCharacters {
+    string expectedResult =
+    "{"
+        "\"testKey1\":\"testLine1\\r\\nTestList2\","
+        "\"testKey2\":\"\\f\\t content\""
+    "}";
+    JsonWriter writer;
+    writer.Add("testKey1", "testLine1\r\nTestList2");
+    writer.Add("testKey2", "\f\t content");
+    string jsonResult = writer.ToString();
+    XCTAssertTrue(
+        jsonResult == expectedResult,
+        "%s",
+        ("Expected result: " + expectedResult + " Result: " + jsonResult).c_str());
+}
+
+- (void) testEscapedCharactersInKeys {
+    string expectedResult =
+    "{"
+        "\"testKeyLine1\\r\\nTestKeyList2\":\"testValue\","
+        "\"\\f\\t key\":32"
+    "}";
+    JsonWriter writer;
+    writer.Add("testKeyLine1\r\nTestKeyList2", "testValue");
+    writer.Add("\f\t key", 32);
+    string jsonResult = writer.ToString();
+    XCTAssertTrue(
+        jsonResult == expectedResult,
+        "%s",
+        ("Expected result: " + expectedResult + " Result: " + jsonResult).c_str());
+}
+
+- (void) testNestedJson {
```

Comment:
maybe add tests where the nested sub-json structure is empty.

Reply:
{"a":123,"b":{},"c":"xyz"} for example.

Reply:
I suggest this because the "add comma if needed" logic is a bit tricky around empty sub objects.

Reply:
Looks like you're not building multiple levels in the same structure, but rather using multiple variable instances, so this might not be a problem you.

Reply:
I will still add a test for this, thanks!